LoopKit/Persistence/PersistenceController.swift

```
@@ -192,3 +224,34 @@ extension PersistenceController: CustomDebugStringConvertible {
         ].joined(separator: "\n")
     }
 }
+
+
+// MARK: - Anchor store/fetch helpers
+
+extension PersistenceController {
+    func storeAnchor(_ anchor: HKQueryAnchor?, key: String) {
+        managedObjectContext.perform {
+            let encoded: Data?
+            if let anchor = anchor {
+                encoded = try? NSKeyedArchiver.archivedData(withRootObject: anchor, requiringSecureCoding: true)
+            } else {
+                encoded = nil
+            }
+            self.updateMetadata(key: key, value: encoded)
+            self.saveInternal()
+        }
+    }
+    
+    func fetchAnchor(key: String, completion: @escaping (HKQueryAnchor?) -> Void) {
+        managedObjectContext.perform {
+            self.fetchMetadata(key: key) { (value) in
+                if let encoded = value as? Data {
+                    let anchor = try? NSKeyedUnarchiver.unarchivedObject(ofClass: HKQueryAnchor.self, from: encoded)
```

[nit] This can silently fail.

---

Added a log; still want the same behavior.